internal/worker/firewaller/firewaller.go

```diff
@@ -1501,171 +1518,279 @@ func (ad *applicationData) scopedContext() (context.Context, context.CancelFunc)
 	return context.WithCancel(ad.catacomb.Context(context.Background()))
 }
 
-// relationLifeChanged manages the workers to process ingress changes for
-// the specified relation.
-func (fw *Firewaller) relationLifeChanged(ctx context.Context, tag names.RelationTag) error {
-	relationKey, err := relation.NewKeyFromString(tag.Id())
+// consumerRelationLifeChanged manages watching for ingress/egress changes on the
+// consuming side. The behavior depends on the local endpoint role:
+// - If requirer: watch local egress changes and publish to remote offering model
+// - If provider: watch remote egress changes and apply ingress locally
+func (fw *Firewaller) consumerRelationLifeChanged(ctx context.Context, relationUUID string) error {
+	shouldStart, rel, err := fw.handleRelationLifeChange(ctx, relationUUID, "consumer")
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if shouldStart {
+		return fw.startConsumerRelation(ctx, rel)
+	}
+	return nil
+}
+
+// offererRelationLifeChanged manages watching for ingress changes on the
+// offering side, where we only watch locally for ingress changes.
+func (fw *Firewaller) offererRelationLifeChanged(ctx context.Context, relationUUID string) error {
+	shouldStart, rel, err := fw.handleRelationLifeChange(ctx, relationUUID, "offerer")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if shouldStart {
+		return fw.startOffererRelation(ctx, rel)
+	}
+	return nil
+}
+
+// handleRelationLifeChange handles the common logic for consumer and offerer
+// relation life changes.
+// It returns a boolean indicating whether to start a new relation watcher,
+// along with the relation tag and details needed to start it.
+func (fw *Firewaller) handleRelationLifeChange(
+	ctx context.Context,
+	relationUUID string,
+	relationType string,
```

Comment:
```suggestion
	relationType RelationType,
```

Make this a type - enum and the values be constants.

Reply:
done.